src/Symfony/Component/Routing/Loader/AnnotationClassLoader.php

```diff
@@ -322,7 +319,7 @@ private function resetGlobals()
         ];
     }
 
-    protected function createRoute($path, $defaults, $requirements, $options, $host, $schemes, $methods, $condition)
+    protected function createRoute(string $path, array $defaults, array $requirements, array $options, ?string $host, array $schemes, array $methods, ?string $condition)
```

Comment:
the constructor of `Route` accepts `string|string[]` for schemes and methods, is it relevant here or is this called always with an array?

Reply:
We're only calling the method with arrays.

https://github.com/symfony/symfony/blob/940eabb1219a54fdd0237e568b44f009e86d5a13/src/Symfony/Component/Routing/Loader/AnnotationClassLoader.php#L151-L152
⬇️ ⬇️ ⬇️ 
https://github.com/symfony/symfony/blob/940eabb1219a54fdd0237e568b44f009e86d5a13/src/Symfony/Component/Routing/Loader/AnnotationClassLoader.php#L205

We could break userland code here though, if that code that extends `AnnotationClassLoader` and calls `createRoute` with strings for `$schemes`and `$method`. I don't know if we need to handle this case.